demo/flutter/lib/main.dart

```diff
@@ -0,0 +1,355 @@
+              alignment: Alignment.topLeft,
+              child: Text(
+                transcriptText,
+                textAlign: TextAlign.left,
+                style: TextStyle(color: Colors.white, fontSize: 20),
+              )
+            )
+          )
+        )
+      );
+  }
+
+  buildLeopardStatusArea(BuildContext context) {
+    return Expanded(
+      flex: 1,
+      child: Container(
+          alignment: Alignment.center,
+          padding: EdgeInsets.only(bottom: 20),
+          child: Text(
+            statusAreaText,
+            style: TextStyle(color: Colors.black),
+          )));
+  }
+
+  buildErrorMessage(BuildContext context) {
+    return Expanded(
+        flex: isError ? 2 : 0,
+        child: Container(
+            alignment: Alignment.center,
+            margin: EdgeInsets.only(left: 20, right: 20),
+            padding: EdgeInsets.all(5),
+            decoration: !isError
+                ? null
+                : BoxDecoration(
+                    color: Colors.red, borderRadius: BorderRadius.circular(5)),
+            child: !isError
+                ? null
+                : Text(
+                    errorMessage,
+                    style: TextStyle(color: Colors.white, fontSize: 18),
+                  )));
+  }
+
+  Widget footer = Expanded(
+      flex: 1,
+      child: Container(
+          alignment: Alignment.bottomCenter,
+          padding: EdgeInsets.only(bottom: 20),
+          child: const Text(
+            "Made in Vancouver, Canada by Picovoice",
+            style: TextStyle(color: Color(0xff666666)),
+          )));
+}
+
+typedef RecordedCallback = Function(double length);
+typedef ProcessErrorCallback = Function(LeopardException error);
+
+class MicRecorder {
+  final VoiceProcessor? _voiceProcessor;
+  int _sampleRate;
+
+  final RecordedCallback _recordedCallback;
+  final ProcessErrorCallback _processErrorCallback;
+  RemoveListener? _removeVoiceProcessorListener;
+  RemoveListener? _removeErrorListener;
+
+  List<int> _pcmData = [];
+
+  static Future<MicRecorder> create(int sampleRate, RecordedCallback recordedCallback, ProcessErrorCallback processErrorCallback) async {
+    return MicRecorder._(sampleRate, recordedCallback, processErrorCallback);
+  }
+
+  MicRecorder._(this._sampleRate, this._recordedCallback, this._processErrorCallback)
+      : _voiceProcessor = VoiceProcessor.getVoiceProcessor(512, _sampleRate) {
+    if (_voiceProcessor == null) {
+      throw LeopardRuntimeException("flutter_voice_processor not available.");
+    }
+    _removeVoiceProcessorListener =
+        _voiceProcessor!.addListener((buffer) async {
+      List<int> frame;
+      try {
+        frame = (buffer as List<dynamic>).cast<int>();
+      } on Error {
+        LeopardException castError = LeopardException(
+            "flutter_voice_processor sent an unexpected data type.");
+        _processErrorCallback(castError);
+        return;
+      }
+
+      _pcmData.addAll(frame);
```

Comment:
I think we should do it same as react native? we saved the pcm to a wav file and used that. Depending on the complexity of flutter we can do it that way, anyways in react native we write our own header file.

Reply:
Sure, I was basing it off Android, but if you think it should be closer to RN I can do that.